service/src/main/java/com/foursquare/service/impl/VenueServiceImpl.java

```diff
@@ -0,0 +1,118 @@
+        Venue venueFound = venueRepository.findByUserIdAndFsId(user.getId(), venue.getFsId());
+
+        if(venueFound != null) {
+            venueRepository.delete(venueFound.getId());
+        } else {
+            throw new VenueException("No delete allowed. Venue doesn't exist with fs id: " + venue.getFsId());
+        }
+    }
+
+    public List<Venue> get() {
+        User user = getCurrentUser();
+        List<Venue> venues = venueRepository.findAlldByUserId(user.getId());
+
+        if (venues.isEmpty()) {
+            throw new VenueException("User with name: " + user.getName() + ", doesn't have any venues");
+        }
+        return venues;
+    }
+
+    private User getCurrentUser() {
+        return userRepository.findByName(SecurityContextHolder.getContext().getAuthentication().getName());
+    }
+
+    private void fillInVenueByDataFromApi(Venue venue) {
+        Optional<JsonNode> venueNode = Optional.ofNullable(searchDao.search(venue.getFsId())).
+                map(venueNodeOptional -> venueNodeOptional.get("response")).map(responseNode -> responseNode.get("venue"));
+
+        venueNode.ifPresent(venueNodeOptional -> {
+            if(DaoResponseVenueValidatior.isValidVenue(venueNodeOptional)) {
+
+                if(venueNodeOptional.get("name") != null) {
+                    venue.setName(venueNodeOptional.get("name").textValue());
+                }
+
+                Optional.ofNullable(venueNodeOptional.get("contact")).
+                    map(contactNode -> contactNode.get("phone")).ifPresent(phoneNode -> venue.setPhone(phoneNode.textValue()));
+
+                Optional.ofNullable(venueNodeOptional.get("location")).
+                    map(locationNode -> locationNode.get("address")).ifPresent(addresssNode ->
+                    venue.setAddress(addresssNode.textValue()));
+
+                Optional.ofNullable(venueNodeOptional.get("categories")).ifPresent(
+                        categoriesNode -> categoriesNode.forEach(categoryNode -> {
+
+                            if(categoryNode.get("id") != null){
+                                Category category = categoryRepository.findByFsId(categoryNode.get("id").textValue());
+
+                                if (category == null) {
+                                    category = new Category();
+                                    category.setFsId(categoryNode.get("id").textValue());
+
+                                    if(categoryNode.get("name") != null) {
+                                        category.setName(categoryNode.get("name").textValue());
+                                    }
+                                    categoryRepository.save(category);
+                                    }
+                                    venue.getCategories().add(category);
+                                    } else {
+                                        throw new VenueException("Category foursquare id doesn't exist");
```

Comment:
It's internal Foursquare error (or not an error at all). We should not reveal it. I suggest just ignore it.